### ISSUE TITLE
Flutter 3.15 crashes on devices running iOS 12 `Check failed: AvailabilityVersionCheck`

### ISSUE THREAD

**Issue by koodimetsa** [flutter/flutter#135317]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Run an app build with Flutter 3.15 on iPad Air or iPhone 5 running iOS 12.
Tested also with master which has the same issue.

### Expected results

App runs like with Flutter 3.10. 

### Actual results

App crashes on startup.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return const Scaffold(body: SafeArea(child: Text('Hello')));
  }
}

```

</details>

### Logs
<details><summary>Logs</summary>

```console
[VERBOSE-3:availability_version_check.cc(39)] Check failed: AvailabilityVersionCheck.
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel beta, 3.15.0-15.2.pre, on macOS 13.6 22G120 darwin-x64, locale en-FI)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.82.2)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!


[✓] Flutter (Channel master, 3.15.0-6.0.pre.71, on macOS 13.6 22G120 darwin-x64, locale en-FI)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.82.2)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by koodimetsa** [comment]
@chinmaygarde This issue seems similar as https://github.com/flutter/flutter/issues/133015.

**Comment by darshankawar** [comment]
Thanks for the report.  I am able to replicate the crash after running a sample app using latest stable and master on iOS simulator 5s (OS 12.4) wherein, it throws below error:

```
2023-09-25 12:36:48.174859+0530 Runner[9728:93045] [FATAL:flutter/shell/platform/darwin/common/availability_version_check.cc(39)] Check failed: AvailabilityVersionCheck.
CoreSimulator 783.5 - Device: iPhone 5s (A639E44F-7021-4491-A87F-80E61D7CF06C) - Runtime: iOS 12.4 (16G73) - DeviceType: iPhone 5s
(lldb) 

```

The above error occurs with and without impeller.

While the same sample app runs properly on iOS simulator 13 pro max (OS 15.2):

<img width="359" alt="Screenshot 2023-09-25 at 12 39 56 PM" src="https://github.com/flutter/flutter/assets/67046386/82e53472-a97d-4871-a89f-9f8250a255ba">

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.13.5, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.13.5 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 12fccda598 (32 hours ago), 2023-09-19 13:56:11 -0700
    • Engine revision bd986c5ed2
    • Dart version 3.1.2
    • DevTools version 2.25.0
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.15.0-7.0.pre.5, on macOS 12.2.1 21D62 darwin-x64,
    locale en-GB)
    • Flutter version 3.15.0-7.0.pre.5 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision be085a9229 (45 minutes ago), 2023-09-25 00:37:04 -0400
    • Engine revision 3ea1174ecf
    • Dart version 3.2.0 (build 3.2.0-191.0.dev)
    • DevTools version 2.28.0-dev.12
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by chinmaygarde** [comment]
This may have been introduced when we tried to work around the [Clang RT version check here](https://github.com/flutter/engine/blob/ef3df40e2e19f091c30f598b083bd9565c0f9698/shell/platform/darwin/common/availability_version_check.cc#L38). @koodimetsa, this is unrelated to the patch you linked.

**Comment by justinekatebe** [comment]
I am also experiencing the same thing. My app is crashing on iOS 12 but working fine on iOS 13+

**Comment by MathieuPierfitte** [comment]
Hi @chinmaygarde, have you started working on a fix for this bug yet? Any idea when the fix might hit stable?

Cheers 🙂

**Comment by appsmark** [comment]
Experiencing same problem while installing app on iPhone 5s running IOS 12.5.7

**Comment by hermuxo** [comment]
Same here. Impacting many people unfortunately. Not sure how to help but happy to.

**Comment by MMMMiller** [comment]
also crashes with Flutter 3.16.0

**Comment by juliansteenbakker** [comment]
As stated in #138711, this issue is breaking iOS 12 in the stable release 3.16.

**Comment by FabioSFernandes** [comment]
Having the same problem here, flutter 3.16... app crashing on "flutter run --release" command

follow some logs to help tracking... 

`[FATAL:flutter/shell/platform/darwin/common/availability_version_check.cc(39)] Check failed: AvailabilityVersionCheck.`

on the device app starts and crashes.. 

`flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.0, on macOS 14.2 23C5030f darwin-x64, locale pt-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.2)
[✓] VS Code (version 1.84.2)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
`
Running it from xcode: 
`dyldbootstrap::start(macho_header...`

Thanks

**Comment by pvddeveloper** [comment]
> Same here. Impacting many people unfortunately. Not sure how to help but happy to in any way I can
> 
> <img alt="image" width="1394" src="https://user-images.githubusercontent.com/7779505/283484866-20732f91-522b-4211-b116-c1e47500786e.png">

Hello, what is the tool you used to check crash? Thanks.

**Comment by hermuxo** [comment]
> Hello, what is the tool you used to check crash? Thanks.

@pvddeveloper It's [App Store Connect ](https://appstoreconnect.apple.com/) > Analytics > Metrics > Usage > Crashes

**Comment by litang0908** [comment]
Same problem on flutter 3.15 and 3.16
Have to downgrade to 3.13.7, it works.

**Comment by skela** [comment]
Also crashing with Flutter 3.16.2

**Comment by juliansteenbakker** [comment]
I can confirm that my app is working on `Channel master, 3.18.0-7.0.pre.88, on macOS 14.1.2 23B92 darwin-x64, locale nl-NL`. The app however does crash after clicking on the home button `(Thread 18: EXC_BAD_ACCESS (code=1, address=0xe04000))`, but i don't think it is related to this issue.

**Comment by appsmark** [comment]
Works again when using master channel 3.18.0-7.0.pre.94 compiling on macOS
14.1.2 and deploying to iPhone 5s software version 12.5.7

**Comment by litang0908** [comment]
Has been fixed at master channel, PR https://github.com/flutter/engine/pull/48624

**Comment by mustafa-707** [comment]
still facing the issue : 

```
2023-12-17 17:21:59.729194+0300 Runner[49534:4547222] [FATAL:flutter/shell/platform/darwin/common/availability_version_check.cc(39)] Check failed: AvailabilityVersionCheck.
CoreSimulator 857.14 - Device: iPhone 6 (02845203-A4C3-45FF-BCBC-33E7383CA404) - Runtime: iOS 12.4 (16G73) - DeviceType: iPhone 6

```

**Comment by ssolon** [comment]
Thumbs up for problem solved in master on an old iPad running iOS 12.5.7.

**Comment by jmagman** [comment]
@zanderso was this also fixed by https://github.com/flutter/engine/pull/48624?  I see you referenced this issue in https://github.com/flutter/flutter/issues/138711#issuecomment-1819831641

**Comment by zanderso** [comment]
Yes, this should also be fixed now.

**Comment by skela** [comment]
will it be included in a future hotfix for 3.16?

**Comment by jmagman** [comment]
> will it be included in a future hotfix for 3.16?

The fix was just released in 3.16.5.
